### ISSUE TITLE
mocha-avoid-only only checking for string literals

### ISSUE THREAD

**Issue by mmilleruva** [microsoft/tslint-microsoft-contrib#436]
I noticed `mochaAvoidOnlyRule.ts` was missing cases where the first argument was not a string literal.  So the following case would be missed:

```typescript
['x', 'y', 'z'].forEach((val) =>  {
   describe.only(val, () => { ...}) 
})
```
I saw that there were test cases specifically written not to catch this case. I believe it is to avoid false positives where `.only` is not from mocha. It might be better to be more aggressive with reporting the issue and people can use https://palantir.github.io/tslint/usage/rule-flags/  to disable false positives.

Perhaps the best approach would be to make this rule a typed rule and check if the return of .only is one of `ExclusiveSuiteFunction` or `ExclusiveTestFunction`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mocha/index.d.ts#L392

**Comment by JoshuaKGoldberg** [comment]
+1 to using type information to fix this false negative! Having it as an optionally typed rule would be best so we still get the existing behavior if no project is provided.

**Comment by reduckted** [comment]
Another case where you can get a false negative is Angular unit tests that use [`fakeAsync`](https://angular.io/api/core/testing/fakeAsync). The code looks like this: 

```ts
it.only('something', fakeAsync(() => {
}));
```

Because the second parameter isn't an arrow function, the rule doesn't consider it to be mocha's `it.only()` function. Making the rule typed would fix this.

**Comment by JoshuaKGoldberg** [comment]
☠️  *It's time!* ☠️ 

Per #876, this repository is no longer accepting feature pull requests. TSLint is being deprecated and we recommend you switch to https://typescript-eslint.io.

Thanks for open sourcing with us, everyone!
